beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -168,3 +171,85 @@ func (s *Server) SubmitAttestations(w http.ResponseWriter, r *http.Request) {
 		http2.WriteError(w, failuresErr)
 	}
 }
+
+// ListVoluntaryExits retrieves voluntary exits known by the node but
+// not necessarily incorporated into any block.
+func (s *Server) ListVoluntaryExits(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "beacon.ListVoluntaryExits")
+	defer span.End()
+
+	sourceExits, err := s.VoluntaryExitsPool.PendingExits()
+	if err != nil {
+		http2.HandleError(w, "Could not get exits from the pool: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	exits := make([]*shared.SignedVoluntaryExit, len(sourceExits))
+	for i, e := range sourceExits {
+		exits[i] = shared.SignedVoluntaryExitFromConsensus(e)
+	}
+
+	http2.WriteJson(w, &ListVoluntaryExitsResponse{Data: exits})
+}
+
+// SubmitVoluntaryExit submits SignedVoluntaryExit object to node's pool
+// and if passes validation node MUST broadcast it to network.
+func (s *Server) SubmitVoluntaryExit(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.SubmitVoluntaryExit")
+	defer span.End()
+
+	if r.Body == http.NoBody {
```

Comment:
good chance to use that new technique 
```
	var jsonRegistrations []*shared.SignedValidatorRegistration
	err := json.NewDecoder(r.Body).Decode(&jsonRegistrations)
	switch {
	case err == io.EOF:
		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
		return
	case err != nil:
		http2.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
		return
	}
	```
 as an example

Reply:
I will rewrite this everywhere